### PR TITLE
Milestone 4: Category filter

### PR DESCRIPTION
**Pull request by shshamim63**
## :white_check_mark: The Pull Request contains the following task:
---

1. Enhances redux store with a filter field.
   ...1. contains current filtered data.
   ...2. Handles All and initial state not selected factor.
2. Contains CategoryFilter component.
   ...1. Components render a selectbox with all the categories.
   ...2. Merges the initial selection All type.
3. Updates filter value on the redux store.
   ...1. Render books that are matched with the current filter value.
   ...2. Implemented a conditional method which should handle to filter the books based on category.

### CODE REVIEW

File: src/reducers/filter.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const bookFilter = (state = '', action) => {
```
Comment (jofer86) [inline src/reducers/filter.js]: We need to make sure to default this to `'All'` as stated in the [Pathwright to this project](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896829/)
image

File: src/utils/filterbooks.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+
+const filteredBooks = (filter, books) => {
+  if (filter !== '' && filter !== 'All') {
```
Comment (jofer86) [inline src/utils/filterbooks.js]: Like it was said on the comment posted in the filter.js file if we default to  `'All'` there would not be necessary for you to pass `''` as a category to filter the books, leaving the filtering defaulted to all should take care of it.

**Review by jofer86 — CHANGES_REQUESTED** [review]
## 👋 👋 Hello Team. Just dropping by to review this milestone, let's take a closer look at your code shall we 👁 👁 👁 

- There is a couple of changes that need to be implemented before you can move on to the next project, those changes are outlined below. Please make sure you tend to those and then request for another code review. Changes are minor, but you should adjust your code accordingly if your changes affect other parts of your app.

--------------------------------------------------------------------
Cheers,
Jorge Fernando Rincón Figueroa.
Technical Support Engineer.
**Slack: @jofer86**

File: src/containers/BookList.js [hunk 3]
```diff
@@ -18,7 +23,7 @@ const BookList = ({ books, removeBook }) => (
       </thead>
       <tbody>
         {
-          books.map(book => (
+          filteredBooks(filter, books).map(book => (
             <Book
               key={book.bookId}
               title={book.title}
```
Comment (jofer86) [inline src/containers/BookList.js]: - We need to make sure that the books are rendered with a conditional statement as stated in [in the Pathwirght to this project](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896829/)
image

**Review by bellom — APPROVED** [review]
## Hi Team :clap: 

**Great work making all the necessary corrections from the previous review.**

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit project completion form.
Goodluck in the next project :slightly_smiling_face:_ 

Best,

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Pull request merged**